app/validators/induction_record_validator.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class InductionRecordValidator < ActiveModel::Validator
+  def validate(record)
+    has_an_induction_record_for_given_provider(record)
+  end
+
+private
+
+  def has_an_induction_record_for_given_provider(record)
+    return if relevant_induction_record?(record)
+
+    record.errors.add(:cpd_lead_provider, I18n.t(:invalid_participant))
+  end
+
+  def relevant_induction_record?(record)
+    InductionRecord
+      .joins(:participant_profile, induction_programme: { school_cohort: [:cohort], partnership: [:lead_provider] })
+      .where(participant_profile: record.participant_profile)
+      .where(induction_programme: { partnerships: { lead_provider: record.cpd_lead_provider.lead_provider } })
+      .where(induction_programme: { school_cohorts: { cohort: Cohort.current } })
```

Comment:
i think there will be a bug if the scope to `Cohort.current` when next cohort kicks in. do we need to include cohort in this query?

Reply:
I've changed this on the branch i'm working on from `Cohort.current` to `Cohort.where(start_year: 2021..)`